### PR TITLE
Metrics check for detecting a process' cpu and memory usage

### PR DESCRIPTION
**Pull request by sivakom**
This sensu check returns the cpu and memory usage (in percent) metrics for a specified process. The output is also structured in a graphite format. The check also takes arguments to check if the usages are surpassing any warning or critical levels.

### CODE REVIEW

**Comment by paulczar** [comment]
What's the goal of this check?  Is there a specific error condition we're looking for?   Can you provide examples of what you want to monitor in ursula with it ?

**Comment by sivakom** [comment]
This check is mainly intended to monitor the various OVN processes, to see if they're exhausting too many resources, in this case memory and cpu, at any point.

**Comment by paulczar** [comment]
perfect :+1:

File: sensu/plugins/metrics-process-usage.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#
+#  metrics-process-usage.py -n <process_name> -w <cpu_warning_pct> -c <cpu_critical_pct> -W <mem_warning_pct> -C <mem_critical_pct> [-s <graphite_scheme>]
+#
+# DESCRIPTION:
+# Finds the pid[s] corresponding to a process name and obtains the necessary
+# cpu and memory usage stats. Returns WARNING or CRITICAL when these stats
+# exceed user specified limits.
+#
+# Code adapted from Jaime Gogo's script in the Sensu Plugins community:
+# https://github.com/sensu-plugins/sensu-plugins-process-checks/blob/master/bin/metrics-per-process.py
```
Comment (omgjlk) [inline sensu/plugins/metrics-process-usage.py:22]: Should we mention the license of the file this came from, and does this file have to remain MIT licensed?

Comment (sivakom) [inline sensu/plugins/metrics-process-usage.py:22]: From the original script:

``` LICENSE:
 Jaime Gago  contact@jaimegago.com
 Released under the same terms as Sensu (the MIT license)
```

Scratch that, that doesn't help. I'm not exactly sure which approach I should be taking.

Comment (omgjlk) [inline sensu/plugins/metrics-process-usage.py:22]: Well, I'm not a lawyer. But this is a derivative work from an MIT licensed file. Presumably this file should also be MIT licensed.

**Comment by sivakom** [comment]
Made this plugin MIT licensed since the script that this is based on is released under those terms as well.

File: sensu/plugins/metrics-process-usage.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+#
+# Released under the same terms as Sensu (the MIT license); see LICENSE
+# for details.
```
Comment (omgjlk) [inline sensu/plugins/metrics-process-usage.py:25]: You point at the LICENSE file, but it only has Apache content. Maybe you should create a new MITLICENSE file to point to.

**Comment by sivakom** [comment]
Added the MIT License, as requested.

**Pull request merged**